src/components/TokenSelect.js

```diff
@@ -30,16 +30,52 @@ const getInsets = () => {
   }
 
   const model = DeviceInfo.getModel();
-  // The values are taken from the list at
-  // https://github.com/react-native-device-info/react-native-device-info/blob/master/ios/RNDeviceInfo/RNDeviceInfo.m#L181-L296
-  if (model.startsWith('iPhone X')) {
-    // This covers IPhone X, XS, XS Max, XR
+  const deviceId = DeviceInfo.getDeviceId();
+
+  // This method will return true for devices with new screen format (introduced in iPhone X)
+  // this is important to get the best padding top here
+  const isNewIphoneScreen = () => {
```

Comment:
Should we move this method to an utils file so we can use elsewhere?

Reply:
I didn't check the implications of [this comment](https://github.com/HathorNetwork/hathor-wallet-mobile/blob/1189aee16e3f7873a0d3fa2ce72339dbd4bd1218/src/components/TokenSelect.js#L87) but i believe not using `SafeAreaView` is the reason we have to manually pad this component.

I couldn't find other instances of this anywhere else, that's why it is contained to this component.
Ideally we should upgrade to use the most recent `SafeAreaView` from react-navigation once we upgrade react-navigation itself, then we can remove this padding. Until then I think we should treat this as a temporary patch and not a feature.

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-mobile/pull/166/commits/eaecc1b4c2104b1f670c862a1a1c63b961cc1710